tests/algorithms/test_blurpool_algorithm.py

```diff
@@ -94,18 +96,39 @@ def test_blurpool(state, blurpool_instance, dummy_logger):
         assert type(state.model.module.pool1) is torch.nn.MaxPool2d
 
 
-def test_blurpool_wrong_event(state, blurpool_instance):
+def test_blurpool_wrong_event(state: State, blurpool_instance: BlurPool):
     assert blurpool_instance.match(Event.BATCH_START, state) == False
 
 
-def test_blurpool_correct_event(state, blurpool_instance):
+def test_blurpool_correct_event(state: State, blurpool_instance: BlurPool):
     assert blurpool_instance.match(Event.INIT, state) == True
 
 
-def test_blurpool_algorithm_logging(state, blurpool_instance, dummy_logger):
+def test_blurpool_algorithm_logging(state: State, blurpool_instance: BlurPool, dummy_logger: Logger):
     blurpool_instance.apply(Event.INIT, state, dummy_logger)
 
     dummy_logger.metric_fit.assert_called_once_with({
         'blurpool/num_blurpool_layers': 1 if blurpool_instance.hparams.replace_maxpools else 0,
         'blurpool/num_blurconv_layers': 1 if blurpool_instance.hparams.replace_convs else 0,
     })
+
+
+def test_blurconv2d_optimizer_params_updated():
+    model = SimpleConvModel()
+    orig_conv = model.conv1
+    assert orig_conv.stride == (2, 2)  # fail fast if test model changes
+    opt = torch.optim.SGD(model.parameters(), lr=.01)
+    apply_blurpool(model, optimizers=opt)
+    new_conv = model.conv1
+    param_list: List[torch.Tensor] = opt.param_groups[0]['params']
+
+    # old params removed
+    assert not surgery._tensor_in(orig_conv.weight, param_list)
+
+    # new params added
+    new_conv2d = new_conv.conv
+    assert isinstance(new_conv2d, torch.nn.Module)
+    new_weight = new_conv2d.weight
+    assert new_weight is not orig_conv.weight
+    assert isinstance(new_weight, torch.Tensor)
+    assert surgery._tensor_in(new_weight, param_list)
```

Comment:
Probably not worth prioritizing right now, but would be nice to split this into a helper func that could work for any surgery method and a method-specific test for each surgery method.